### PR TITLE
Use in memory cache when adding inBoundGroupSession

### LINKED ISSUE #4011: RageShake | Messages are displayed as unabled to decrypt then decrypted a few seconds later
https://github.com/matrix-org/element-android-rageshakes/issues/26594

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Fixes #4011

### CODE REVIEW

File: changelog.d/4011.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Messages are displayed as unable to decrypt then decrypted a few seconds later #4011
```
Comment (bmarty) [inline changelog.d/4011.bugfix:1]: remove issue id in the content please.

**Review by bmarty — COMMENTED** [review]
What about changing also https://github.com/vector-im/element-android/blob/main/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/MXOlmDevice.kt#L648 which is called at keybackup restoration or at manual keys import?

File: changelog.d/4011.bugfix (hunk 1 above)
Comment (BillCarsonFr) [reply changelog.d/4011.bugfix]: done

**Review by bmarty — APPROVED** [review]

**Comment by bmarty** [comment]
Sorry, ktlint is not happy, can you fix it please?

**Pull request merged**